app/com/linkedin/drelephant/exceptions/util/ExceptionInfo.java

```diff
@@ -15,56 +15,94 @@
  */
 
 package com.linkedin.drelephant.exceptions.util;
-
+import lombok.Data;
 /**
  * Class to store exception information
  */
-public class ExceptionInfo {
+@Data
+public class ExceptionInfo implements Comparable<ExceptionInfo> {
 
   private int exceptionID;
+  private Integer weightOfException;
   private String exceptionName;
   private String exceptionStackTrace;
   private ExceptionSource exceptionSource;
 
+  private String exceptionTrackingURL;
+
+  /**
+   * Added for serialize and deserialize into JSON
+   */
+  public ExceptionInfo(){
+
+  }
+
+  @Override
+  public int compareTo(ExceptionInfo o) {
+    return o.weightOfException.compareTo(this.weightOfException);
+  }
+
   public enum ExceptionSource {DRIVER, EXECUTOR, SCHEDULER}
 
   public ExceptionInfo(int exceptionID, String exceptionName, String exceptionStackTrace,
-      ExceptionSource exceptionSource) {
+      ExceptionSource exceptionSource, int weightOfException, String exceptionTrackingURL) {
     this.exceptionID = exceptionID;
     this.exceptionName = exceptionName;
     this.exceptionStackTrace = exceptionStackTrace;
     this.exceptionSource = exceptionSource;
+    this.weightOfException = weightOfException;
+    this.exceptionTrackingURL = exceptionTrackingURL;
   }
 
   //TODO: Currently this has not been used . But the idea to have ID of two excpetion same
   //if they are simillar , so that we can remove simillar exception from the exception fingerprinting
   //system
+  // Getter and setters are used by GSON/Jackson for serialize and deserialize into JSON
 
-  public int getExceptionID() {
-    return exceptionID;
-  }
 
-  public void setExceptionID(int exceptionID) {
-    this.exceptionID = exceptionID;
-  }
+  //TODO : Use exception source to prioritize  the exception
 
-  public String getExceptionName() {
-    return exceptionName;
+  @Override
```

Comment:
nit: Can `toString` annotation from Lombok be feasible to use here?

Reply:
done